drivers/src/mailbox.rs

```diff
@@ -70,10 +69,44 @@ impl Mailbox {
     /// * 'MailboxRecvTxn' - Object representing a receive operation
     pub fn try_start_recv_txn(&self) -> Option<MailboxRecvTxn> {
         let mbox = mbox::RegisterBlock::mbox_csr();
-        match mbox.status().read().mbox_fsm_ps() {
-            MboxFsmE::MboxExecuteUc => Some(MailboxRecvTxn::default()),
-            _ => None,
+        if mbox.status().read().mbox_fsm_ps().mbox_execute_uc() {
+            match mbox.status().read().status() {
+                MboxStatusE::CmdBusy => Some(MailboxRecvTxn::default()),
+                _ => None,
+            }
+        } else {
+            None
+        }
+    }
+}
+
+pub struct MailboxTxn;
+
+impl MailboxTxn {
+    pub(crate) fn enqueue(buf: &[u8]) -> CaliptraResult<()> {
```

Comment:
This should probably just be a top-level function.

Reply:
Ack. Will do.

Reply:
Moved runtime mailbox usage outside of this driver